### ISSUE TITLE
Move StorageEngines and PropertyBackends out of `core`

### ISSUE THREAD

**Issue by dnorman** [synestheticsystems/ankurah#12]
As a forcing function to help ensure that Ankurah does not ossify around a fixed set of types/storage backends, we should move the existing property Backends out of `core` (except the minimum needed for certain tests)

One relevant intuition I have about this is that Ankurah should ideally NOT have a standard set of abstract materialized types.
It may temporarily have an `enum Value` as a short term expedience, but this should not be maintained. Instead, it should be replaced by a pairwise mapping between the native types of each Property Backend, and the native types of each StorageEngine.

In the case of Sled, those types are whatever you want to {de}serialize to/from Vec<u8>, which is flexible, but not very interesting.
In the case of Postgres, there are a variety of native types that simultaneously would be lovely to support, and also have no business influencing a "unified" type system.

For example, the interim Value type I am using as a short term expedience now:

```
pub enum Value {
    String(String),
    Integer(i64),
    Float(f64),
    Boolean(bool),
}
```

Sure, this is good enough for lots of use cases, but it's also far too blunt an instrument for a lot of other use cases which I'd prefer not to handicap, like different kinds of sets, or geo types, or vector types, etc. Sure, those use cases could just shove everything into a string or binary type, but then you lose a lot of potential functionality, and make for a worse developer experience.

Conversely, adding a zillion subtly different types to such a unified enum makes no sense either, because there will be too many, and they'll end up having weird impedance mismatches with the underlying storage engine, assuming they get added at all, which is more likely.

So in summary, in the course of implementing this change, we should include a trait that allows for the implementation of type conversions, usually from PropertyBackend to StorageEngine, such that the maximum semantic fidelity is preserved. This may seem obnoxious to have to implement this redundantly, but it shouldn't be that bad in practice.

**Comment by Aceeri** [comment]
Storage Engines:
- [ ] Sled
- [ ] Postgres
- [X] IndexDB

Property Backends:
- [ ] Yrs
- [ ] Last Write Wins (LWW)
- [ ] Positive-Negative Counter (PNCounter)

**Comment by dnorman** [comment]
Phase 2: #267 (public PropertyBackend contract plus conformance kit) is the enabling step for moving backends out of core; once the contract and kit exist, out-of-tree backends become testable against executable laws. Tracking: #269.
